### PR TITLE
updated readme.md with instructions for compiling for EPS32

### PR DESCRIPTION
**Pull request by v923z**
Should fix https://github.com/v923z/micropython-ulab/issues/273

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,40 +106,126 @@ in question.
...
+ESPIDF_VER=9e70825d1e1cbf7988cf36981774300066580ea7
+
+# Download and prepare the SDK
+git clone https://github.com/espressif/esp-idf.git $BUILD_DIR/esp-idf
+cd $BUILD_DIR/esp-idf
+git checkout $ESPIDF_VER
+git submodule update --init --recursive # get idf submodules
+pip install -r ./requirements.txt # install python reqs
+```
+
```
Comment (CallumJHays) [inline README.md:184]: Duplicate paragraph
```suggestion
```

**Comment by v923z** [comment]
@CallumJHays Thanks, Cal!

**Comment by CallumJHays** [comment]
I just realized that these instructions are missing the step of cloning the repo into the `$BUILD_DIR/ulab`.

Might want to add this in as the second-last step:

```bash
git clone https://github.com/v923z/micropython-ulab $BUILD_DIR/ulab
```

**Comment by v923z** [comment]
> I just realized that these instructions are missing the step of cloning the repo into the `$BUILD_DIR/ulab`.

@CallumJHays Indeed. Thanks!

**Comment by mdaeron** [comment]
I just managed to compile for TinyPICO on MacOS with the script at the bottom of this message. Everything seems to work fine but, just in case it's useful, I though I would report that I get the two following warnings:

````
In file included from ../../../esp-idf/components/freertos/include/freertos/portable.h:89:0,
                 from ../../../esp-idf/components/freertos/include/freertos/FreeRTOS.h:105,
                 from ../../../esp-idf/components/cxx/cxx_guards.cpp:22:
../../../esp-idf/components/soc/esp32/include/soc/cpu.h: In function 'void cpu_configure_region_protection()':
../../../esp-idf/components/soc/esp32/include/soc/cpu.h:77:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i = 0; i < sizeof(pages_to_protect)/sizeof(pages_to_protect[0]); ++i) {
````

````
In file included from ../../../esp-idf/components/freertos/include/freertos/portable.h:89:0,
                 from ../../../esp-idf/components/freertos/include/freertos/FreeRTOS.h:105,
                 from ../../../esp-idf/components/nvs_flash/src/nvs_platform.hpp:19,
                 from ../../../esp-idf/components/nvs_flash/src/nvs_api.cpp:18:
../../../esp-idf/components/soc/esp32/include/soc/cpu.h: In function 'void cpu_configure_region_protection()':
../../../esp-idf/components/soc/esp32/include/soc/cpu.h:77:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i = 0; i < sizeof(pages_to_protect)/sizeof(pages_to_protect[0]); ++i) {
````

Installation script:

````sh
clear

HERE=`pwd`

echo "\n--- GET CROSS-COMPILER ---\n"
cd $HERE
wget https://dl.espressif.com/dl/xtensa-esp32-elf-osx-1.22.0-80-g6c4433a-5.2.0.tar.gz
tar -zxvf xtensa-esp32-elf-osx-1.22.0-80-g6c4433a-5.2.0.tar.gz

echo "\n--- CLONE MICROPYTHON ---\n"
cd $HERE
git clone https://github.com/micropython/micropython.git

# cd micropython/ports/esp32
# make

echo "\n--- CLONE ESP-IDF ---\n"
cd $HERE
git clone https://github.com/espressif/esp-idf.git
cd esp-idf
git checkout 9e70825d1e1cbf7988cf36981774300066580ea7

echo "\n--- ESP-IDF SUBMODULE UPDATE ---\n"
git submodule update --init

echo "\n--- CLONE ULAB ---\n"
cd $HERE
git clone https://github.com/v923z/micropython-ulab.git ulab
 
echo "\n--- MICROPYTHON SUBMODULE UPDATE ---\n"
cd $HERE/micropython
git submodule update --init
 
echo "\n--- MAKE MPY-CROSS ---\n"
cd $HERE/micropython/mpy-cross
make
 
echo "\n--- MAKE TINYPICO ---\n"
cd $HERE/micropython/ports/esp32
export PATH="../../../xtensa-esp32-elf/bin:$PATH"
export ESPIDF=../../../esp-idf
make BOARD=TINYPICO USER_C_MODULES=../../../ulab all
 
echo "\n--- FLASH TINYPICO ---\n"
cd $HERE/micropython/ports/esp32/build-TINYPICO
esptool.py --chip esp32 --port /dev/tty.usb* erase_flash
esptool.py --chip esp32 --port /dev/tty.usb* --baud 460800 write_flash -z 0x1000 firmware.bin

echo "\n--- CONNECT TO TINYPICO ---\n"
rshell -p /dev/tty.usb* -b 115200
````

**Comment by v923z** [comment]
@mdaeron It seems to me that these warnings are totally unrelated to `ulab`: the offending line is in one of the ESP32 components. I bet that you would get the same warnings, even if you compiled without `ulab`.

**Comment by mdaeron** [comment]
> @mdaeron It seems to me that these warnings are totally unrelated to `ulab`: the offending line is in one of the ESP32 components. I bet that you would get the same warnings, even if you compiled without `ulab`.

Correct, sorry. I am, as ever, quite lost when it comes to compiling C code.

**Comment by v923z** [comment]
@mdaeron Absolutely no problem, don't worry!

**Pull request merged**